orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy

```diff
@@ -141,15 +141,8 @@ class BakeStage implements StageDefinitionBuilder {
 
     @Nonnull
     TaskResult execute(@Nonnull StageExecution stage) {
-      def bakeInitializationStages = stage.execution.stages.findAll {
-        it.parentStageId == stage.parentStageId && it.status == ExecutionStatus.RUNNING
-      }
-
-      // FIXME? (lpollo): I don't know why we do this, but we include in relatedStages all parallel bake stages in the
-      //  same pipeline, not just the ones related to the stage passed to this task, which means this list actually
-      //  contains *unrelated* bake stages...
       def relatedBakeStages = stage.execution.stages.findAll {
-        it.type == PIPELINE_CONFIG_TYPE && bakeInitializationStages*.id.contains(it.parentStageId)
+        it.type == PIPELINE_CONFIG_TYPE && stage.id == it.parentStageId
```

Comment:
Do we want to add `it.status == ExecutionStatus.RUNNING`  to the filter, since it was present before?

Reply:
I suspect the status==RUNNING check *might* have been a dirty workaround to sort of limit the related bake stages that were found

With the check for parent stage Id I think the current code is correct

Reply:
Yes, as far as I was able to figure out, the RUNNING check was part of the problem. For instance, if you had multiple parallel bake stages, and one or more of those bakes was already baked (causing Rosco to reuse the existing bake), only those reused bakes would be present in the `outputs`. The proper bakes would be stripped away.